src/server/zset_family.cc

```diff
@@ -2791,32 +2809,43 @@ bool MembersOfAllNeighbors(ConnectionContext* cntx, string_view key, const GeoHa
   }
 
   // get all the matching members and add them to the potential result list
-  auto cb = [&](Transaction* t, EngineShard* shard) {
-    return OpRanges(range_specs, t->GetOpArgs(shard), key);
-  };
-  OpResult<vector<ScoredArray>> result_arrays =
-      cntx->transaction->ScheduleSingleHopT(std::move(cb));
-  if (result_arrays.status() == OpStatus::WRONG_TYPE) {
-    cntx->SendError(kWrongTypeErr);
-    return false;
-  } else if (result_arrays.status() == OpStatus::KEY_NOTFOUND) {
-    cntx->SendError("Member not found");
-    return false;
+  vector<OpResult<vector<ScoredArray>>> result_arrays;
+  if (write_mode) {
+    auto cb = [&](Transaction* t, EngineShard* shard) {
```

Comment:
I would move the transaction related code to the caller (GeoSearchStoreGeneric). Basically, the division to the functions is sub-optimal imho.